File: starter_kits/Scala/hlt/MapCell.scala
package hlt

class MapCell(val position: Position, var halite: Int) {
  var ship: Ship = null
  var structure: Entity = null

  def isEmpty: Boolean = ship == null && structure == null

  def isOccupied: Boolean = ship != null

  def hasStructure: Boolean = structure != null

  def markUnsafe(ship: Ship): Unit = this.ship = ship
}

File: starter_kits/Kotlin/hlt/MapCell.kt
package hlt

class MapCell(val position: Position, var halite: Int) {
    var ship: Ship? = null
    var structure: Entity? = null

    val isEmpty: Boolean
        get() = ship == null && structure == null

    val isOccupied: Boolean
        get() = ship != null

    fun hasStructure(): Boolean {
        return structure != null
    }

    fun markUnsafe(ship: Ship) {
        this.ship = ship
    }
}
